### ISSUE TITLE
Block comment ignores strings

### ISSUE THREAD

**Issue by PavelVozenilek** [delta-lang/delta#58]
This failed:

```
/*
void main() {
    println("Hello /* world"); // <<< open block comment in a string
}
*/

void main() {
    println("Hello world");
}
```

with error:

```
main.delta:1:1: error: unterminated block comment
/*
^

```

I noticed, that in file `parser/lex.cpp` the function `Lexer::readBlockComment` doesn't handle strings.

**Comment by emillaine** [comment]
Good catch. But I don't this is feasible to implement, because comments may contain arbitrary text, making it nearly impossible to reliably detect if the `/*` is inside a "string" or not. For example:

```delta
/* Some explanation here.
var c = '"'; /* The ascii value of `"` is 34.
                More explanation. */
*/
```

Here the `/*` is enclosed between two `"` characters on the same line, but it's actually part of a nested block comment, and therefore should not be ignored.

I tested with Swift, which also has nested comments, and it doesn't seem to handle strings inside block comments in a special way either.

**Comment by PavelVozenilek** [comment]
I noticed it when reading lexer'code. One way would be to lex everything first (including comments start/stop), and then to filter comments away.

**Comment by emillaine** [comment]
That wouldn't work because comments can contain text that doesn't lex properly. For example the text `"\"` is valid in a comment, but lexing would treat is as an unterminated string literal (since the last quote is escaped).
